tests/Unit/Domain/Test_ElementDistribution.cpp

```diff
@@ -3,306 +3,810 @@
+  // 1D
+  test_proc_retrieval_impl(element_weight, domain_creator_1d, 1);
+  test_proc_retrieval_impl(element_weight, domain_creator_1d, 5);
+  test_proc_retrieval_impl(element_weight, domain_creator_1d, 10,
+                           std::unordered_set<size_t>{4, 6});
+  test_proc_retrieval_impl(element_weight, domain_creator_1d, 33,
+                           std::unordered_set<size_t>{7});
+
+  // 2D
+  test_proc_retrieval_impl(element_weight, domain_creator_2d, 1);
+  test_proc_retrieval_impl(element_weight, domain_creator_2d, 5);
+  test_proc_retrieval_impl(element_weight, domain_creator_2d, 20,
+                           std::unordered_set<size_t>{4, 20});
+  test_proc_retrieval_impl(element_weight, domain_creator_2d, 54,
+                           std::unordered_set<size_t>{0, 1});
+
+  // 3D
+  test_proc_retrieval_impl(element_weight, domain_creator_3d, 1);
+  test_proc_retrieval_impl(element_weight, domain_creator_3d, 12);
+  test_proc_retrieval_impl(element_weight, domain_creator_3d, 73,
+                           std::unordered_set<size_t>{5, 8, 9, 75});
+  test_proc_retrieval_impl(element_weight, domain_creator_3d, 500,
+                           std::unordered_set<size_t>{100});
+}
+
+void test(const domain::ElementWeight element_weight) {
+  // Test inputs
+
+  // 1D, single block
+  const auto lattice_1d = make_domain_creator<1>(
+      "AlignedLattice:\n"
+      "  BlockBounds: [[0.0, 1.0]]\n" +
+      std::string{"  IsPeriodicIn: [false]\n"} +
+      "  InitialGridPoints: [6]\n"
+      "  InitialLevels: [4]\n"
+      "  RefinedLevels: []\n"
+      "  RefinedGridPoints: []\n"
+      "  BlocksToExclude: []\n");
+
+  // 2D
+  const auto lattice_2d = make_domain_creator<2>(
+      "AlignedLattice:\n"
+      "  BlockBounds: [[0.0, 0.3], [0.0, 0.8, 2.5, 4.9]]\n" +
+      std::string{"  IsPeriodicIn: [false, false]\n"} +
+      "  InitialGridPoints: [4, 5]\n"
+      "  InitialLevels: [2, 3]\n"
+      "  RefinedLevels: []\n"
+      "  RefinedGridPoints: []\n"
+      "  BlocksToExclude: []\n");
+
+  // 3D
+  const auto binary_compact_object_creator =
+      TestHelpers::test_option_tag<domain::OptionTags::DomainCreator<3>,
+                                   Metavariables<3, true, false>>(
+          create_option_string(true, true, true, false, false, 0, 0, 0, false));
+
+  if (element_weight == domain::ElementWeight::Uniform) {
+    test_uniform_cost_function();
+    test_uniform_element_distribution(*lattice_1d, *lattice_2d,
+                                      *binary_compact_object_creator);
+  } else {
+    test_weighted_cost_function(element_weight);
+    test_weighted_element_distribution(element_weight, *lattice_1d, *lattice_2d,
+                                       *binary_compact_object_creator);
   }
+
+  test_proc_retrieval(element_weight, *lattice_1d, *lattice_2d,
+                      *binary_compact_object_creator);
 }
 }  // namespace
+
+SPECTRE_TEST_CASE("Unit.Domain.ElementDistribution", "[Domain][Unit]") {
```

Comment:
I know you like to be very thorough with your tests, but this is timing out. Is it possible to reduce the runtime of the tests without affecting coverage? E.g. since you've factored out some of the z-curve code, can this test be simpler because you can assume your z-curve tests are working?

Reply:
Yes, I tried trimming down the time with a fixup on another branch and it's still timing out on some CI checks, so I'm still iterating on it but will push a fixup when I get it to run without any timeouts